packages/big-design/src/components/Worksheet/Cell/Cell.tsx

```diff
@@ -82,15 +85,20 @@ const InternalCell = <T extends WorksheetItem>({
     setSelectedCells([cell]);
   }, [cell, rowIndex, setSelectedCells, setSelectedRows]);
 
-  const renderedCell = useMemo(
-    () =>
-      isEditing ? (
-        <Editor cell={cell} isEdited={isEdited} onBlur={handleBlur} onKeyDown={handleKeyDown} />
-      ) : (
-        <Small color="secondary70">{`${value}`}</Small>
-      ),
-    [cell, handleBlur, handleKeyDown, isEdited, isEditing, value],
-  );
+  const renderedCell = useMemo(() => {
+    switch (type) {
+      case 'select':
+        return <SelectEditor cell={cell} isEdited={isEdited} onChange={handleChange} options={options} />;
+      // Default to TextEditor
+      default:
+        return isEditing ? (
+          <TextEditor cell={cell} isEdited={isEdited} onBlur={handleBlur} onKeyDown={handleKeyDown} />
+        ) : (
+          // In case of NaN casting to string
+          <Small color="secondary70">{value.toString()}</Small>
+        );
+    }
+  }, [cell, handleBlur, handleChange, handleKeyDown, isEdited, isEditing, options, type, value]);
```

Comment:
The previous logic of returning an `Editor` proved to be too tricky for Typescript and/or wasn't the right approach. I moved the switch to the `Cell` for a more transparent approach, and it is typed correctly.